public/app/features/browse-dashboards/components/RestoreModal.tsx

```diff
@@ -27,8 +27,13 @@ export const RestoreModal = ({
   const numberOfDashboards = selectedDashboards.length;
 
   useEffect(() => {
-    if (Object.entries(dashboardOrigin).length === 1 && dashboardOrigin[selectedDashboards[0]] !== 'general') {
-      setRestoreTarget(dashboardOrigin[selectedDashboards[0]]);
+    const origins = Object.values(dashboardOrigin);
+    if (origins.length === 1) {
+      setRestoreTarget(origins[0]);
+    } else if (origins.length > 1) {
+      if (origins.filter((originalLocation) => originalLocation !== origins[0]).length === 0) {
+        setRestoreTarget(origins[0]);
+      }
```

Comment:
It took me a while to realize what was happening here. Could we turn this into one `if`?

```
if (origins.every((originalLocation) => originalLocation === origins[0])) {
  setRestoreTarget(origins[0]);
}
```

And also add some sort of comment to explain why?

Reply:
+1 ☝️

Reply:
Good point. I cleaned up now and it looks so much better.